addons/website_quote/controllers/main.py

```diff
@@ -79,75 +75,64 @@ def view(self, order_id, pdf=None, token=None, message=False, **post):
                         'type': 'form',
                         'alias_usage': _('If we store your payment information on our server, subscription payments will be made automatically.'),
                         'partner_id': order.partner_id.id,
-                    },
-                    context=render_ctx)
+                    })
         return request.website.render('website_quote.so_quotation', values)
 
     @http.route(['/quote/accept'], type='json', auth="public", website=True)
     def accept(self, order_id, token=None, signer=None, sign=None, **post):
-        order_obj = request.registry.get('sale.order')
-        order = order_obj.browse(request.cr, SUPERUSER_ID, order_id)
-        if token != order.access_token:
-            return request.website.render('website.404')
-        if order.require_payment:
+        order = request.env['sale.order'].sudo().browse(order_id)
+        if token != order.access_token or order.require_payment:
             return request.website.render('website.404')
         if order.state != 'sent':
             return False
-        attachments=sign and [('signature.png', sign.decode('base64'))] or []
-        order_obj.action_confirm(request.cr, SUPERUSER_ID, [order_id], context=request.context)
+        attachments = sign and [('signature.png', sign.decode('base64'))] or []
```

Comment:
`cond and stmt1 or stmt2` is kind of a old way to program. We should use `stmt1 if cond else stmt2 ` to look more trendy ;)